sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4541,6 +4541,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_RETAIN_FRACTION_DIGITS_FIRST =
+    buildConf("spark.sql.legacy.decimalLeastCommonType.retainFractionDigitsFirst")
+      .internal()
+      .doc("When set to true, we will try to retain the fraction digits first rather than " +
+        "integral digits, when getting a least common type between decimal types, and the " +
+        "result decimal precision exceeds the max precision.")
```

Comment:
```suggestion
      .doc("When set to true, we will try to retain the fraction digits first rather than " +
        "integral digits as prior Spark 4.0, when getting a least common type between decimal types, and the " +
        "result decimal precision exceeds the max precision.")
```